### PR TITLE
feat(GitHub): Retry retrieval of Piper download URL

### PR DESCRIPTION
**Pull request by jliempt**
Implements retries for GitHub server issues when retrieving Piper's download URL.

There are two other GET requests for which retries are already present:
- `downloadTool()` in `github.ts` to actually download the binary
- Piper's `getDefaults` in `config.ts`

### CODE REVIEW

File: src/fetch.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import { info } from '@actions/core'
+
+export async function wait (delay: number): Promise<string> {
+  return await new Promise((resolve) => setTimeout(resolve, delay))
+}
+
+export async function fetchRetry (url: string, tries = 5, baseDelayMS = 1000): Promise<Response> {
+  let attempt = 0
+
+  while (tries > 0) {
```
Comment (CCFenner) [inline src/fetch.ts:10]: Do not modify `tries` and compare with `attempt`s.
```suggestion
  while (tries > attempt) {
```

File: src/fetch.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      return response
+    }
+
+    info(`Error while fetching ${url}: ${response.statusText}`)
+    if (!isRetryable(response.status)) {
+      break
+    }
+
+    attempt += 1
+    tries -= 1
```
Comment (CCFenner) [inline src/fetch.ts:22]:
```suggestion
```

File: src/fetch.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    if (tries > 0) {
```
Comment (CCFenner) [inline src/fetch.ts:24]:
```suggestion
    if (tries > attempt) {
```

File: src/fetch.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      const delayTime = baseDelayMS * Math.pow(2, attempt - 1)
+      info(`Retrying ${tries} more time(s)...`)
```
Comment (CCFenner) [inline src/fetch.ts:26]:
```suggestion
      info(`Retrying ${tries-attempt} more time(s)...`)
```

**Review by CCFenner — APPROVED** [review]

File: src/fetch.ts (hunk 1 above)
Comment (jliempt) [reply src/fetch.ts]: Done

File: src/fetch.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+      await wait(delayTime)
+    }
+  }
+  return await Promise.reject(new Error(`Error fetching ${url}`))
+}
+
+function isRetryable (code: number): boolean {
+  switch (code) {
+    case 408: // Request Timeout
+    case 429: // Too Many Requests
```
[the remark this answers is not in the document]
Comment (jliempt) [reply src/fetch.ts]: With the exponential backoff I'd say yes. But I removed it just in case. We can investigate if we ever run into this response.

**Pull request merged**